packages/components/src/components/menusection/menusection.component.ts

```diff
@@ -0,0 +1,94 @@
+import { CSSResult, html } from 'lit';
+import { queryAssignedElements } from 'lit/decorators.js';
+import { TAG_NAME as MENUITEMRADIO_TAGNAME } from '../menuitemradio/menuitemradio.constants';
+import { TAG_NAME as MENUITEMCHECKBOX_TAGNAME } from '../menuitemcheckbox/menuitemcheckbox.constants';
+import type MenuItemRadio from '../menuitemradio/menuitemradio.component';
+import { DEFAULTS, ARIA_CHECKED_STATES } from './menusection.constants';
+import { Component } from '../../models';
+
+/**
+ * `mdc-menusection` is a container element used to group a set of menu items.
+ *
+ * This component supports a mix of `menuitem`, `menuitemcheckbox`, and `menuitemradio` components.
+ *
+ * - If multiple `menuitemradio` components are slotted into the section, the group enforces a single-selection rule:
+ *   only one radio item can be selected at a time.
+ * - If `menuitemcheckbox` components are included, their checked state can be toggled independently.
+ *
+ * @tagname mdc-menusection
+ *
+ * @slot - Default slot for inserting `menuitem`, `menuitemcheckbox`, or `menuitemradio`
+ */
+class MenuSection extends Component {
+  /**
+   * Query assigned `menuitemradio` elements from the default slot.
+   * These elements are used to enforce single-selection behavior.
+   *
+   * @internal
+   */
+  @queryAssignedElements({ selector: `${MENUITEMRADIO_TAGNAME}` })
```

Comment:
```suggestion
  @queryAssignedElements({ selector: `${MENUITEMRADIO_TAGNAME}:not([disabled])` })
```

If the menu item is disabled, we should not set it to true or false.